test/e2e/storage/testsuites/volumemode.go

```diff
@@ -67,12 +66,9 @@ func (t *volumeModeTestSuite) defineTests(driver TestDriver, pattern testpattern
 		config      *PerTestConfig
 		testCleanup func()
 
-		cs     clientset.Interface
-		ns     *v1.Namespace
-		sc     *storagev1.StorageClass
-		pvc    *v1.PersistentVolumeClaim
-		pv     *v1.PersistentVolume
-		volume TestVolume
+		cs       clientset.Interface
+		ns       *v1.Namespace
+		resource *volumeModeTestResource
```

Comment:
you may be able to save one layer if you embed `genericVolumeTestResource` directly.

Reply:
Actually, I moved ```cs``` and ```ns``` to ```volumeModeTestResource``` once and moved back here, because they aren't used in ```createVolumeModeTestResource``` and ```cleanupResource```. 

If your intention is different to above, or it's better to move there even above situation. Please point me out. I will update.

Reply:
What I meant was it is possible to just have `genericVolumeTestResource` directly in `local`, like this:
```
type local struct {
  config *PerTestConfig
  ...
  genericVolumeTestResource
}
```

Then you don't need to add a `.resource` everytime you need to access the pvc, pv, sc.

Reply:
Thank you for your detailed explanation. I will fix as suggested.